crates/ruff/src/rules/flake8_return/visitor.rs

```diff
@@ -12,9 +13,10 @@ pub struct Stack<'a> {
     pub elifs: Vec<&'a Stmt>,
     pub refs: FxHashMap<&'a str, Vec<Location>>,
     pub non_locals: FxHashSet<&'a str>,
-    pub assigns: FxHashMap<&'a str, Vec<Location>>,
+    pub assigns: FxHashMap<&'a str, Vec<Range>>,
     pub loops: Vec<(Location, Location)>,
     pub tries: Vec<(Location, Location)>,
+    pub assign_values: FxHashMap<&'a str, Expr>,
```

Comment:
Can this be `FxHashMap<&'a str, &'a Expr>`? Would that let you avoid the clone?

Reply:
I will try this.